crypto/test/test_util.cc

```diff
@@ -166,27 +167,33 @@ size_t createTempFILEpath(char buffer[PATH_MAX]) {
 }
 
 size_t createTempDirPath(char buffer[PATH_MAX]) {
-  char pathname[PATH_MAX];
-  char tempdir[PATH_MAX];
+  char temp_path[PATH_MAX];
+  uint8_t random_bytes[8]; // 64 bits of randomness is sufficient
 
-  if (0 == GetTempPathA(PATH_MAX, pathname)) {
+  // Get the temporary path
+  if (0 == GetTempPathA(PATH_MAX, temp_path)) {
     return 0;
   }
 
-  // Generate a unique name using Windows API
-  if (0 == GetTempFileNameA(pathname, "awslctestdir", 0, tempdir)) {
+  if (!RAND_bytes(random_bytes, sizeof(random_bytes))) {
     return 0;
   }
 
-  // Delete the file that GetTempFileNameA created
-  DeleteFileA(tempdir);
+  int written = snprintf(buffer, PATH_MAX, "%s\\awslctest_%02x%02x%02x%02x%02x%02x%02x%02x",
+                        temp_path,
+                        random_bytes[0], random_bytes[1], random_bytes[2], random_bytes[3],
+                        random_bytes[4], random_bytes[5], random_bytes[6], random_bytes[7]);
```

Comment:
NP: It would be cleaner to declare `random_bytes` as a `uint64_t`:
```
  uint64_t random_bytes[8]; // 64 bits of randomness is sufficient
...
  if (!RAND_bytes(&random_bytes, sizeof(random_bytes))) {
    return 0;
  }

  int written = snprintf(buffer, PATH_MAX, "%s\\awslctest_%" , temp_path, random_bytes);
```